src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -57,9 +67,15 @@ public static SAMRecord applyAlignment( final String readName, final byte[] base
                         }
                     }
                 }
-                bases = Arrays.copyOfRange(bases, alignment.getSeqStart(), alignment.getSeqEnd());
+            }
+            if (tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ||
+                    tmpCigar.getLastCigarElement().getOperator() == CigarOperator.H) {
+                final int leftHardClip = tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ? tmpCigar.getFirstCigarElement().getLength() : 0;
```

Comment:
You don't need to calculate the length of the hard clips -- the alignment tells you the proper bounds.  The old code was fine, and simpler.
```
bases = Arrays.copyOfRange(bases, alignment.getSeqStart(), alignment.getSeqEnd());
if ( quals.length != 0 ) quals = Arrays.copyOfRange(quals, alignment.getSeqStart(), alignment.getSeqEnd());
```